### DISCUSSION TITLE
WebKit build fails due to limited RAM per CPU core

### DISCUSSION THREAD

**Discussion by blakelloyd-coderus** [Igalia/meta-webkit#178] [Q&A] [answered]
Currently trying to build wpe for a rpi3 32bit (have tried both options with hardcoded values and all sorts) but I keep getting to the same error:

```
gcc  ...                   -fdebug-prefix-map=/home/jenkins/workspace/branch/poky/build/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/wpewebkit/2.26.2-r0/recipe-sysroot-native=  -fvisibility-inlines-hidden  -mthumb -mfpu=neon-vfpv4 -mfloat-abi=hard -mcpu=cortex-a7 -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/jenkins/workspace/branch/poky/build/tmp/work/cortexa7t2hf-neon-vfpv4-poky-linux-gnueabi/wpewebkit/2.26.2-r0/recipe-sysroot -fno-strict-aliasing -fno-exceptions -fno-rtti -DNDEBUG -fPIC   -std=c++17 -MD -MT Source/WebCore/CMakeFiles/WebCore.dir/__/__/DerivedSources/WebCore/unified-sources/UnifiedSource-c57e08af-7.cpp.o -MF Source/WebCore/CMakeFiles/WebCore.dir/__/__/DerivedSources/WebCore/unified-sources/UnifiedSource-c57e08af-7.cpp.o.d -o Source/WebCore/CMakeFiles/WebCore.dir/__/__/DerivedSources/WebCore/unified-sources/UnifiedSource-c57e08af-7.cpp.o -c DerivedSources/WebCore/unified-sources/UnifiedSource-c57e08af-7.cpp
| ninja: build stopped: subcommand failed.
| WARNING: exit code 1 from a shell command.
```

**Comment by clopez** [comment]
Since the WebKit build its parallelized the error will not be on the last line.
You need to find the complete log file (the log.do_compile file) and then look on that file for lines containing the word error (case insensitive)

**Comment by blakelloyd-coderus** [comment]
Ahh, well that makes way more sense!

`FAILED: Source/JavaScriptCore/CMakeFiles/JavaScriptCore.dir/dfg/DFGSpeculativeJIT.cpp.o `

`arm-poky-linux-gnueabi-g++: fatal error: Killed signal terminated program cc1plus
compilation terminated.`

Seems to be the actual issue

**Comment by clopez** [comment, marked as answer]
> arm-poky-linux-gnueabi-g++: fatal error: Killed signal terminated program cc1plus compilation terminated.

This usually mean you ran out of ram and the kernel killed the gcc process. You need at least 2GB of RAM per CPU vcore to build WebKit. If you don't have enough then try to lower the number of build threads.

For example you can set in local.conf
```
# Default to setting automatically based on cpu count
BB_NUMBER_THREADS = "4"

# Default to setting automatically based on cpu count
PARALLEL_MAKE = "-j 4"
```

With that number (4) you would need 8GB of RAM to build WebKit

**Comment by blakelloyd-coderus** [comment]
Ahh ok, I thought I was just missing the error that was there!

It was running on 12 cores with 10GB RAM so that may well have been the issue, currently rebuilding. Thanks!

**Comment by blakelloyd-coderus** [comment]
Well that seems to fix it, thanks for the help!
